spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,316 @@
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-tuple-exotic-objects-ownpropertykeys">
+        <h1>[[OwnPropertyKeys]] ( )</h1>
+        <p>When the [[OwnPropertyKeys]] internal method of a Record exotic object _R_ is called, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _keys_ be a new empty List.
+          1. Assert _T_ is Tuple exotic object.
+          1. Let _len_ be _T_.[[TupleLength]].
+          1. For each integer _i_ starting with 0 such that _i_ &lt; _len_ in ascending order, do
+            1. Add ! ToString(_i_) to _keys_.
+          1. Return _keys_
+        </emu-alg>
+      </emu-clause>
+
+      
+      <emu-clause id="sec-tuplecreate" aoid="TupleCreate">
+        <h1>TupleCreate ( _value_ )</h1>
+        <p>This abstract operation TupleCreate takes a _value_ argument. It is used to specify the creation of new Tuple exotic objects. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: Type(_value_) is Tuple.
+            1. Assert: Type(_value_.[[Sequence]]) is List.
+            1. Let _T_ be ! MakeBasicObject(&laquo; [[TupleData]], [[Prototype]] &raquo;).
+            1. Let _T_.[[TupleLength]] be _value_.[[Sequence]].length
+            1. Set _T_'s essential internal methods to the definitions specified in <emu-xref href="#sec-tuple-exotic-objects"></emu-xref>.
+            1. Set _T_.[[Prototype]] to *null*.
+            1. Set _T_.[[TupleData]] to _value_.
+            1. Return _T_
+            
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-isvalidtupleindex-r-p" aoid="IsValidTupleIndex">
+        <h1>IsValidTupleIndex ( _T_, _numericIndex_ )</h1>
+        <p>This abstract operation IsValidTupleIndex takes arguments _T_ and _numericIndex_. It performs the following steps when called:</p>
+        <emu-alg>
+          1. Assert: _T_ is an Object that has [[TupleData]] internal slot.
+          1. Assert: Type(_numericIndex_) is an integer.
+          1. If _numericIndex_ is *-0*, return *false*
+          1. If _numericIndex_ &lt; 0 or index &gte; _T_.[[TupleLength]], return *false*.
+          1. Return *true*.
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-tupleget-r-p" aoid="TupleGet">
+        <h1>TupleGet ( _T_, _numericIndex_ )</h1>
+        <p>This abstract operation TupleGet takes arguments _T_ and _numericIndex_. It performs the following steps when called:</p>
+        <emu-alg>
+          1. Assert: _T_ is an Object that has [[TupleData]] internal slot.
+          1. Assert: Type(_numericIndex_) is an integer.
+          1. Let _tup_ be _T_.[[TupleData]].
+          1. If ! IsValidTupleIndex(_numericIndex_) is *false*, return *undefined*.
```

Comment:
In this case, return `~empty~` instead. (Or, assert it's true and expect callers to call IsValidTupleIndex.)